control-plane-operator/controllers/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -389,6 +389,19 @@ func reconcileAWSEndpointService(ctx context.Context, awsEndpointService *hyperv
 			}
 			return err
 		}
+
+		if aws.StringValue(output.VpcEndpoints[0].ServiceName) != awsEndpointService.Status.EndpointServiceName {
+			log.Info("conflicting endpoint found", "VPCEndpointServiceName", aws.StringValue(output.VpcEndpoints[0].ServiceName), "AWSEndpointServiceName", awsEndpointService.Status.EndpointServiceName)
+			if _, err := ec2Client.DeleteVpcEndpointsWithContext(ctx, &ec2.DeleteVpcEndpointsInput{
+				VpcEndpointIds: []*string{output.VpcEndpoints[0].VpcEndpointId},
+			}); err != nil {
+				return fmt.Errorf("error deleting AWSEndpoint: %w", err)
+			}
+
+			// Once the VPC Endpoint is deleted, we need to send an error in order to reexecute the reconcilliation
+			return fmt.Errorf("VPCEndpoint deleted %s", aws.StringValue(output.VpcEndpoints[0].VpcEndpointId))
```

Comment:
I'd expect the error message to describe the problem rather than the action taken.

Reply:
Make sense, changing the message accordingly